src/services/annotations-parser.js

```diff
@@ -17,23 +17,85 @@ export function parseAnnotationSets(manifest, canvasIndex) {
     const annotations = canvas.annotations;
     const duration = Number(canvas.duration);
 
-    if (annotations?.length > 0 && annotations[0].type === 'AnnotationPage') {
-      annotations.map((annotation) => {
-        if (annotation.type === 'AnnotationPage') {
-          let annotationSet = { label: getLabelValue(annotation.label) };
-          if (annotation.items?.length > 0) {
-            annotationSet.items = parseAnnotationItems(annotation.items, duration);
-          } else {
-            annotationSet.url = annotation.id;
-          }
-          annotationSets.push(annotationSet);
-        }
+    annotationSets = parseAnnotationPage(annotations, duration);
+    return { canvasIndex, annotationSets };
+  } else {
+    return null;
+  }
+};
+
+/**
+ * Fetch and parse linked AnnotationPage json file
+ * @param {String} url URL of the linked AnnotationPage .json
+ * @param {Number} duration Canvas duration
+ * @returns {Object} JSON object for the annotations
+ * 
+ */
+export async function parseExternalAnnotationPage(url, duration) {
```

Comment:
Where does this method get used?  Should it get called in `parseAnnotationPages`?

Reply:
This method is supposed to get called from the UI component when a selection is made. I tested it locally with some dummy code in the interface. I didn't commit that code, as it was not complete code plus I didn't want to mix the parsing stuff with the UI.